packages/x-components/src/plugins/__tests__/x-plugin-mixin.spec.ts

```diff
@@ -26,54 +28,91 @@ describe('testing $x component API global mixin', () => {
   });
 
   describe('testing origin', () => {
-    it("doesn't include origin in the $x.emit metadata by default", () => {
-      const listener = jest.fn();
-
-      componentInstance.vm.$x.on('UserIsTypingAQuery', true).subscribe(listener);
-      componentInstance.vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
-
-      expect(listener).toHaveBeenCalledTimes(1);
-      expect(listener).toHaveBeenCalledWith({
-        eventPayload: 'So awesome, much quality, such skill',
-        metadata: { moduleName: null }
-      });
-    });
-
-    it('overrides the origin in the $x.emit metadata', () => {
-      const listener = jest.fn();
-
-      @Component({
-        template: `<div><slot/></div>`,
-        provide: {
-          origin: 'origin-test'
-        }
-      })
-      class Provider extends Vue {}
+    interface RenderOriginComponentOptions {
```

Comment:
Can we document these interfaces please?

Reply:
Done